src/app/geo-accession-download/geo-accession-download.component.ts

```diff
@@ -0,0 +1,46 @@
+import {Component, Input} from '@angular/core';
+import {LoaderService} from "@shared/services/loader.service";
+import {SaveFileService} from "@shared/services/save-file.service";
+import {BrokerService} from "@shared/services/broker.service";
+import {AlertService} from "@shared/services/alert.service";
+
+@Component({
+  selector: 'app-geo-accession-download',
+  templateUrl: './geo-accession-download.component.html',
+  styleUrls: ['./geo-accession-download.component.css']
+})
+
+export class GeoAccessionDownloadComponent {
+  @Input() geoAccession;
+
+  constructor(
+    private loaderService: LoaderService,
+    private saveFileService: SaveFileService,
+    private brokerService: BrokerService,
+    private alertService: AlertService,
+  ) { }
+
+
+  downloadSpreadsheetUsingGeo() {
+    const geo_accession = this.geoAccession;
+
+    if (!geo_accession) {
+      console.log('Unable to download the spreadsheet as GEO accession is not specified')
```

Comment:
should this be an alert?

Reply:
No, this is intended. This is just being defensive. The button is also hidden anyway. This handler function will never get called.

Reply:
cc @yusra-haider